package NKikimrBlobStorage;
option java_package = "ru.yandex.kikimr.proto";

// sets the parameters of VDisk
message TVDiskConfig {
    optional uint64 FreshBufSizeLogoBlobs = 10;
    optional uint64 FreshBufSizeBarriers = 11;
    optional uint64 FreshBufSizeBlocks = 12;
    optional uint64 FreshCompThresholdLogoBlobs = 13;
    optional uint64 FreshCompThresholdBarriers = 14;
    optional uint64 FreshCompThresholdBlocks = 15;

    optional bool FreshReadReadyMode = 20;
    optional bool FreshUseDreg = 21;

    optional bool AllowKeepFlags = 40;

    optional uint32 HullCompLevel0MaxSstsAtOnce = 45;
    optional uint32 HullCompSortedPartsNum = 46;

    optional uint32 ReplInterconnectChannel = 50;

    optional bool BarrierValidation = 60;
    optional bool EnableOverseerLsnReporting = 61; // deprecated
};

// organizes hierarchy of VDisk configs: VDisk config may have a base config,
// local settings override base settings
message TVDiskKind {
    enum EVDiskKind {
        Default = 0;
        // reserved for tests
        Test1 = 1;
        Test2 = 2;
        Test3 = 3;
        // optimized for tablet log
        Log = 10;
        LocalMode = 11;
        Extra2 = 22;
        Extra3 = 23;
        Extra4 = 24;
        Extra5 = 25;
    };

    optional EVDiskKind Kind = 1;
    optional EVDiskKind BaseKind = 2;
    optional TVDiskConfig Config = 3;
};

// all VDisk kinds are enumerated here
message TAllVDiskKinds {
    repeated TVDiskKind VDiskKinds = 1;
};

message TIncrHugeConfig {
    optional uint32 MinHugeBlobInBytes = 1 [default = 500000];
    optional uint32 MinCleanChunks = 2 [default = 8];
    optional uint32 MinAllocationBatch = 3 [default = 4];
    optional uint32 UnalignedBlockSize = 4 [default = 50000];
    optional uint32 MaxInFlightWrites = 5 [default = 5];
};
